tools/ci_build/github/azure-pipelines/post-merge-jobs.yml

```diff
@@ -35,24 +29,16 @@ stages:
       runTests: true
       buildJava: false
       buildNodejs: false
+      # The public ADO project
+      ${{ if eq(variables['System.CollectionId'], 'f3ad12f2-e480-4533-baf2-635c95467d29') }}:
+        ort_build_pool_name: 'Win-CPU-2019'
+      # The private ADO project
+      ${{ if eq(variables['System.CollectionId'], 'bc038106-a83b-4dab-9dd3-5a41bc58f34c') }}:
+        ort_build_pool_name: 'Win-CPU-2021'
```

Comment:
Can we set this in one place instead of cut-and-pasting?

Does it matter if ort_build_pool_name does not get set because neither of the values for System.CollectionId were found?

Reply:
I don't find a better way. Azure DevOps's templating is not very flexible. It can only be used in certain places. 

For our team, "System.CollectionId" can only be the two values. One for https://dev.azure.com/onnxruntime, another for https://dev.azure.com/aiinfra.  The other teams generally would not reuse our YAML files. Their builds start with our build.py.

Reply:
Can we use the `variables:` section at the top of the yaml file like this?

https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#set-variables-in-pipeline

Reply:
Variables are dynamic.  The occurrences of "${{" are templates. Templates are static and more like preprocessing macros. Templates can be interpreted and processed offline, but variables only when the task is running. So, if there is anything wrong with a template syntax, we know it when we queue a build or run an offline validator. But variables we cannot.

Reply:
Let me try it.